### PR TITLE
Made Receiver default owner for Closure.

### PR DESCRIPTION
**Pull request by emilwestergren**
Fixes https://github.com/cogneco/ooc-kean/issues/1062
@sebastianbaginski @simonmika peer review

### CODE REVIEW

**Comment by simonmika** [comment]
I think this doesn't work as it requires a default constructor for covers. We probably need to change the order of the `Owner` enum.

**Comment by marcusnaslund** [comment]
@simonmika is right, this will not work, unfortunately.

**Comment by emilwestergren** [comment]
Oh that's right. So I change the order of the enum instead?

**Comment by emilwestergren** [comment]
Is this how it's supposed to work:
```ooc
closure := (function as Closure)
closure = closure take()
closure free(Owner Sender)
```

**Comment by emilwestergren** [comment]
It gets pretty messy to use this in general since you need to cast back and forth from `Closure` and `Func`.
```ooc
f := func {} as Closure
f = f take()
(f as Func)()
f free(Owner Sender)
```

**Comment by emilwestergren** [comment]
Updated.

**Comment by simonmika** [comment]
This looks right. I agree with the messiness. maybe there is a way to extend `Func` although I'm doubting. Let us go ahead anyhow as this is better then what we are having today.

**Comment by simonmika** [comment]
Peer review done.

**Comment by marcusnaslund** [comment]
Won't other things be `Owner Receiver` instead of `Owner Unknown` by default? Will that cause any sideeffects? I will have to test this rather closely.

File: test/sdk/OwnedClosureTest.ooc [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+			f := function2 as Closure
+			instance1 := MyClass new()
+			expect(f owner == Owner Receiver)
+			f = f give()
+			f free(Owner Sender)
+			expect(f owner == Owner Receiver)
+			(f as Func(MyClass))(instance1)
+			instance1 apply(f as Func(MyClass))
+			expect(instance1 content, is equal to(12))
+			instance1 free()
```
Comment (marcusnaslund) [inline test/sdk/OwnedClosureTest.ooc]: I assume you split these up afterwards? Anyway, just call them `instance` without the `1`.

**Comment by emilwestergren** [comment]
Updated.

**Pull request merged**